article/app/services/dotcomponents/AMPPicker.scala

```diff
@@ -4,7 +4,8 @@ import common.Logging
 import controllers.ArticlePage
 import implicits.Requests._
 import model.PageWithStoryPackage
-import model.liveblog._
+import model.liveblog.BlockElement
+import model.dotcomrendering.pageElements._
```

Comment:
@nicl @MatthewJWalls Does this make sense?

Reply:
(We've already discussed this so just noting for posterity!)

Unfortunately not in this case. There are three models here:

* liveblog block element model
* atom thrift models from CAPI
* page element model sent to DCR

The question is: which of these should the AMP picker use?

Initially we used the liveblog model as it was at a similar granularity to the page element ones we were creating. However, this is not ideal, because it's not a perfect 1-2-1 mapping.

We can't use the (DCR) page element model because we don't model things we don't yet support (by definition), so we can't match against those here. In addition, it requires building the entire DCR model before we might simply reject the request, which seems wasteful.

So probably the right answer is to use the atom model! As then we can just mirror the match logic in PageElement.scala. There is still no guarantee of a perfect 1-2-1 mapping, as some atoms we might partially support. But I think it's good enough for our purposes.

I'll raise a PR!

Reply:
Or happy to pair/you to do it.